dev/src/codewind/project/Requester.ts

```diff
@@ -159,8 +162,67 @@ namespace Requester {
 
         // return doProjectRequest(project, url, body, Requester.post, "Build");
         const buildMsg = Translator.t(STRING_NS, "build");
+        if (project.connection.remote) {
+            const localPath = MCUtil.fsPathToContainerPath(project.localPath);
+            Log.i(`Clearing existing content for ${project.name} from ${project.connection.host}`);
+            await requestClear(project);
+            Log.i(`Copying updated files from ${localPath} to ${project.connection.host}`);
+            await requestUploadsRecursively(project.connection, project.id, localPath, localPath);
+        } else {
+            Log.i(`Local build from local file system at ${project.localPath}`);
+        }
         await doProjectRequest(project, ProjectEndpoints.BUILD_ACTION, body, Requester.post, buildMsg);
-        // await requestValidate(project, true);
+    }
+
+    export async function requestClear(project: Project): Promise<void> {
+        const msg = Translator.t(STRING_NS, "clear");
+        await doProjectRequest(project, ProjectEndpoints.CLEAR, {}, Requester.post, msg, true);
+    }
+
+
+    export async function requestUploadsRecursively(connection: Connection, projectId: any, inputPath: string, parentPath: string): Promise<void> {
+        Log.i(`requestUploadsRecursively for ${projectId} at ${inputPath}`);
+        const files = fs.readdirSync(inputPath);
+
+        for (const f of files) {
+            const currentPath = `${inputPath}/${f}`;
+            // Log.i("Uploading " + currentPath);
+            const stats = fs.statSync(currentPath);
+            if (stats.isFile()) {
+                try {
+                    await remoteUpload(connection, projectId, currentPath, parentPath);
+                } catch (err) {
+                    Log.d(err);
+                }
+            } else if (stats.isDirectory()) {
+                await requestUploadsRecursively(connection, projectId, currentPath, parentPath);
+            }
+        }
+    }
+
+    async function remoteUpload(connection: Connection, projectId: string, filePath: string, parentPath: string): Promise<any> {
```

Comment:
what kind of response does this return? we shouldn't use `any`, if we don't care about the response then use `void`